src/desktop/mainwindow.cpp

```diff
@@ -1198,6 +1211,87 @@ void MainWindow::onCanvasSaved(const QString &errorMessage)
 		close();
 }
 
+void MainWindow::promptForSaveOnReset(const drawdance::CanvasState &canvasState)
+{
+	const char *objectName = "resetsessionmessagebox";
+	bool saveOrPromptInProgress = m_doc->isSaveInProgress() ||
+		findChild<QMessageBox *>(objectName, Qt::FindDirectChildrenOnly);
+	if(!saveOrPromptInProgress) {
+		QMessageBox::Icon icon;
+		QString message;
+		if(m_doc->isCompatibilityMode()) {
+			icon = QMessageBox::Warning;
+			message = tr(
+				"The session has been reset. Since this is a Drawpile 2.1 "
```

Comment:
In testing, what are some different things that you have noticed?

Reply:
Drawpile 2.1 uses 8 bit pixels, while Drawpile 2.2 uses 15 bits. The lack of bits leads to artifacts when dealing with low opacity.

Indirect (aka wash aka opacity) drawing mode is just plain broken in Drawpile 2.1, it uses the wrong blend mode, so it ends up being the same as direct (aka build up aka flow) mode, just with some pointless creation and merging of sublayers at the beginning and end of every stroke. So pretty much every stroke drawn in indirect mode involving opacity will look different between them.

There's also minute differences in float rounding and such, but those also exist within Drawpile 2.1 depending on which code paths Qt decides to take and in Drawpile 2.2 depending on how your SIMD is doing. But the differences are so minute that it's far beneath human perception even after drawing for a long time.

Reply:
Since the biggest thing sounds like the broken indirect mode is it possible to just implement the broken way again for sessions that are joined in compatibility mode?